mpags-cipher.cpp

```diff
@@ -0,0 +1,244 @@
+// mpags-cipher.cpp
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+int main( int argc, char* argv[]){
+
+    const std::vector<std::string> cmdLineArgs{argv, argv+argc};
+
+    // Creating a boolean variable to tell us when help us requested.
+    bool helpRequested{false};
+
+    // Creating a boolean variable to tell me when the version is requested.
+    bool versionRequested{false};
+
+    // Creating a boolean to check if the input file was given and initialising the input file variable
+    bool input_fileGiven{false};
+    std::string input_file{""};
+
+    // Creating a boolean to check if the output file was given and initialising the output file variable
+    bool output_fileGiven{false};
+    std::string output_file{""};
```

Comment:
You don't actually need the boolean variables, because the strings are initially set to be empty, so subsequently you can just check whether they are non-empty to see if a filename has been specified